src/bob/config.clj

```diff
@@ -0,0 +1,76 @@
+;   This file is part of Bob.
+;
+;   Bob is free software: you can redistribute it and/or modify
+;   it under the terms of the GNU Affero General Public License as published by
+;   the Free Software Foundation, either version 3 of the License, or
+;   (at your option) any later version.
+;
+;   Bob is distributed in the hope that it will be useful,
+;   but WITHOUT ANY WARRANTY; without even the implied warranty of
+;   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+;   GNU Affero General Public License for more details.
+;
+;   You should have received a copy of the GNU Affero General Public License
+;   along with Bob. If not, see <http://www.gnu.org/licenses/>.
+
+(ns bob.config
+  (:require [clojure.edn :as edn]
+            [environ.core :as env :refer [env]]
+            [schema.core :as s]
+            [taoensso.timbre :as log]
+            [bob.api.schemas :as schema]))
+
+(defn int-from-env
+  [key default]
+  (try
+    (Integer/parseInt (get env key (str default)))
+    (catch Exception _ default)))
+
+(defn read-config-file
+  "read config from .bob.conf"
+  []
+  (try
+    (when-let [env-file (clojure.java.io/file "./.bob.conf")]
+      (when (.exists env-file)
+        (edn/read-string (slurp env-file))))
+    (catch Exception e
+      (log/warnf "Failed to parse .bob.conf: %s" e))))
+
+(defn merge-configs
+  "Merging two configs"
+  [confa confb]
+  (try
+    (merge-with merge confa confb)
```

Comment:
not sure if this will raise an exception 🤔

Reply:
Maybe we can also go with more deeper merges with something like
https://dnaeon.github.io/recursively-merging-maps-in-clojure/ ?
We may get more maps in maps? 😛

Reply:
exception is raised here when the config read from file is just a string or vec for example. then it returns still a config from defaults/env/props but warns on the file.

you want more configs to be merged? what do you have in mind when thinking about deeper merges? anything in particular?

Reply:
should be okay for now, was keeping the option open for future things but would be easy to add in then.